dask/array/overlap.py

```diff
@@ -190,7 +206,7 @@ def trim_internal(x, axes, boundary=None):
         ilist = []
         for j, d in enumerate(bd):
             if bdy != 'none':
-                d = d - axes.get(i, 0) * 2
+                d = d - axes.get(i, 0) * 2  #TODO add support for assymmetrical axes
```

Comment:
Why does this case need special handling?

Reply:
I think that there will need to be additional work to support cases where (-/+) tuples are passed as overlap depth dimensions. In cases where boundary is not None.

Reply:
So I must admit, I'm a little concerned that this functionality doesn't work for the bulk of boundary cases. At a bare minimum, we should raise an error at the outset when given an asymmetric overlap with padding other than `"none"`. Though I'd feel much better if we were able to support these other cases. What are the difficulties with implementing this?

Reply:
I do not foresee any difficulties.  Only time constraints.  The current change is only implements support in overlap_internal, which technically does not implement boundaries.  In that regards, this particular pull request is self consistent and passes all tests.  Technically then, I would guess the TODO's can be removed.  If it were merged as is, I would probably follow up with an additional PR with changes to overlap() and supporting tests, which would included the boundaries.

The aforementioned can of of course be added to this PR, if there is consensus that the approach is valid and accepted.  Thank you, this is very good feedback.

Reply:
Alright could we then raise an exception if users supply an asymmetric overlap with a border condition? Also let's raise an issue describing the functionality still needed to support asymmetric overlap for all other cases. Does that sound reasonable?

Reply:
I think that's a great idea.

Reply:
Are you comfortable doing those things? If so, then I think we can merge once that is done.

Reply:
Yes, I am working it.  One question.  The default boundary=None -> boundary={i:'reflect'} for i in ndim.  In other words kwarg None, leads to relfected boundaries rather than none, which is a bit counterintuitive.  I started off fixing that, but am now having to repair all of the tests which assume boundaries have been added when boundary=None.   Do you think it would be worthwhile to continue to fix coerce_boundaries to coerce to 'none' in this PR, or just expect the user of assymetric boundaries to know that boundaries=None is not the same as boundaries={i:'none'}

Reply:
yes, great